pepper-apis/docs/specification/src/endpoints/user.studies.activity.answer.yml

```diff
@@ -0,0 +1,44 @@
+parameters:
+  - in: path
+    name: user
+    required: true
+    description: the participant's guid
+    schema:
+      type: string
+  - in: path
+    name: study
+    required: true
+    description: the study's unique identifier
+    schema:
+      type: string
+  - in: path
+    name: instance
+    required: true
+    description: the activity instance guid
+    schema:
+      type: string
+  - in: path
+    name: answerGuid
+    required: true
+    description: the answer guid
+    schema:
+      type: string
+
+delete:
+  operationId: deleteAnswer
```

Comment:
Regarding idea about `DELETE` /answer` endpoint: How about we allow a client to patch a `null` for answer `value` in the PATCH? I propose that if server sees a `null` then it will somehow "delete" the FileAnswer and mark the FileUpload as "outdated" or similar so the bucket file can be cleaned up.

Related question: What happens when client PATCH a new file upload to a file question that already has a file answer? I assume new file upload should somehow "replace" the old one. What happens to the old FileAnswer then? I propose the old FileAnswer be deleted somehow and it's associated FileUpload be marked "outdated" for cleanup. This way, the behavior here will be similar to if client patched `null`, so I think we'll have some consistency?

Reply:
> Regarding idea about `DELETE /answer` endpoint: How about we allow a client to patch a `null` for file answer `value` in the PATCH? I propose that if server sees a `null` then it will somehow "delete" the FileAnswer and mark the FileUpload as "outdated" or similar so the bucket file can be cleaned up.
> 
Deleting answer with a null PATCH value is consistent with other answer types, so I agree we can drop `DELETE` (the DELETE operation I believe was necessary when PATCH was being used for the purpose of getting the upload URL).

Also agree about handling "deleted" files.

> Related question: What happens when client PATCH a new file upload to a file question that already has a file answer? I assume new file upload should somehow "replace" the old one. What happens to the old FileAnswer then? I propose the old FileAnswer be deleted somehow and its associated FileUpload be marked "outdated" for cleanup. This way, the behavior here will be similar to if client patched `null`, so I think we'll have some consistency?

I think that we should aim to reproduce the semantics of PATCH that we are using for existing questions. So if the activity instance is still writeable (i.e., not readonly), then previous file answers should be able to be replaced.

The proposed implementation of file answer is to have a reference (foreign key) to a file upload object. The file uploads will have a status. 
There is also supposed to be a job that cleans out unused uploads. This same job could also remove unreferenced upload records in the database and corresponding files in bucket.

So, yeah, a null for answer or a new file answer ought to trigger similar handling for existing/previous file uploads.

Reply:
Sounds good.

Reply:
0d300fcd
<img width="756" alt="Screen Shot 2021-01-05 at 3 42 29 PM" src="https://user-images.githubusercontent.com/29984380/103696869-165aa180-4f6d-11eb-8f29-7a58b96b94e9.png">